define _mm256_set_m128i for gcc compilers < 8.0

During installation in Python 3.6.9 with `gcc 7.5.0` on Ubuntu 18.04:
```bash
sudo pip3 install tomotopy
```

I received the following error:
```
src/python/../Labeling/../Utils/EigenAddonOps.hpp:79:8: error: ‘_mm256_set_m128i’ was not declared in this scope
        u = _mm256_set_m128i(
            ^~~~~~~~~~~~~~~~
    src/python/../Labeling/../Utils/EigenAddonOps.hpp:79:8: note: suggested alternative: ‘_mm256_set_epi8’
        u = _mm256_set_m128i(
            ^~~~~~~~~~~~~~~~
            _mm256_set_epi8
```

The computer that has the following CPU information:
```
doug@doug-desktop:~$ lscpu
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              8
On-line CPU(s) list: 0-7
Thread(s) per core:  2
Core(s) per socket:  4
Socket(s):           1
NUMA node(s):        1
Vendor ID:           AuthenticAMD
CPU family:          21
Model:               1
Model name:          AMD FX(tm)-8120 Eight-Core Processor
Stepping:            2
CPU MHz:             1419.705
CPU max MHz:         3100.0000
CPU min MHz:         1400.0000
BogoMIPS:            6242.06
Virtualization:      AMD-V
L1d cache:           16K
L1i cache:           64K
L2 cache:            2048K
L3 cache:            8192K
NUMA node0 CPU(s):   0-7
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf pni pclmulqdq monitor ssse3 cx16 sse4_1 sse4_2 popcnt aes xsave avx lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs xop skinit wdt fma4 nodeid_msr topoext perfctr_core perfctr_nb cpb hw_pstate ssbd ibpb vmmcall arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold
```

I believe that this can be resolved by [defining `_mm256_set_m128i`](https://stackoverflow.com/a/32630658/2829487)

--- a/src/Utils/EigenAddonOps.hpp
+++ b/src/Utils/EigenAddonOps.hpp
@@ -33,6 +33,12 @@ namespace Eigen
 	}
 }
 
+#ifdef __GNUC__
+#if __GNUC__ < 8
+#define _mm256_set_m128i(v0, v1)  _mm256_insertf128_si256(_mm256_castsi128_si256(v1), (v0), 1)
+#endif
+#endif
+
 #ifdef EIGEN_VECTORIZE_AVX
 #include <immintrin.h>
 #include "avx_gamma.h"

